Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/LoadTestDataServiceImpl.java

```diff
@@ -48,64 +54,182 @@
+                LOG.error("DAO fail to save patient basic-info.");
+                throw new LoadTestDataException("Patient basic-info cannot be save.");
             }
-
         }
         else {
-            throw new LoadTestDataException("Patient-Personname is required when trying to save-patient.");
+            LOG.info("Patient basic-info is required");
+            throw new LoadTestDataException("Patient basic-info is required when trying to save-patient.");
+        }
+        return actionResult;
+    }
+
+    @Override
+    public List<Personname> getAllPersonnamesBy(Long patientId) {
+        return personnameDAO.findRecords(patientId);
+    }
+
+    @Override
+    public List<Address> getAllAddressesBy(Long patientId) {
+        return addressDAO.findRecords(patientId);
+    }
+
+    @Override
+    public List<Identifier> getAllIdentiersBy(Long patientId) {
+        return identifierDAO.findRecords(patientId);
+    }
+
+    @Override
+    public List<Phonenumber> getAllPhonenumbersBy(Long patientId) {
+        return phonenumberDAO.findRecords(patientId);
+    }
+
+    @Override
+    public boolean deletePersonname(Personname personname) {
+        return personnameDAO.delete(personname);
+    }
+
+    @Override
+    public boolean deleteAddress(Address address) {
+        return addressDAO.delete(address);
+    }
+
+    @Override
+    public boolean deleteIdentifier(Identifier identifier) {
+        return identifierDAO.delete(identifier);
+    }
+
+    @Override
+    public boolean deletePhonenumber(Phonenumber phonenumber) {
+        return phonenumberDAO.delete(phonenumber);
+    }
+
+    @Override
+    public Personname getPersonnameBy(Long personnameId) {
+        return personnameDAO.read(personnameId);
+    }
+
+    @Override
+    public Address getAddressBy(Long addressId) {
+        return addressDAO.read(addressId);
+    }
+
+    @Override
+    public Identifier getIdentifierBy(Long identifierId) {
+        return identifierDAO.read(identifierId);
+    }
+
+    @Override
+    public Phonenumber getPhonenumberBy(Long phonenumberId) {
+        return phonenumberDAO.read(phonenumberId);
+    }
+
+    @Override
+    public boolean savePersonname(Personname personname) throws LoadTestDataException {
+        boolean actionResult = false;
+        if(personname.getPatient() != null){
+            if (HelperUtil.isId(personname.getPersonnameId())) {
+                actionResult = personnameDAO.update(personname);
+            }else{
+                actionResult = personnameDAO.create(personname);
+            }
+            if (!actionResult) {
+                LOG.error("DAO fail to save personname.");
+                throw new LoadTestDataException("Personname fail to save to database.");
+            }
+        }
+        return actionResult;
+    }
+
+    @Override
+    public boolean saveAddress(Address address) throws LoadTestDataException {
+        boolean actionResult = false;
```

Comment:
To be consistent with other DML operations in GenericDAOImpl.java, initialize the actionResult to true and change the logic accordingly. Refer GenericDAOImpl.update method.